### PR TITLE
Add more columns to CSV File

### PR DESCRIPTION
**Pull request by dannyjgibson**
Add more columns to CSV output. And adds a function to get desired tracks data from associated tracks. Completes ask on [JIRA L2REQ-20](https://jwplayer.atlassian.net/browse/L2REQ-20)

### CODE REVIEW

**Review by astraussner — APPROVED** [review]
Looks good!

File: examples/video_list_to_csv.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import math
 
 import jwplatform
```
Comment (boyntoni) [inline examples/video_list_to_csv.py:8]: nit but can remove this

File: examples/video_list_to_csv.py [hunk 2]
```diff
@@ -61,18 +61,41 @@ def make_csv(secret, site_id, path_to_csv=None, result_limit=1000, query_params=
...
+def get_captions(api_client, site_id, media_id):
+    captions_response = api_client.request(
+        method='GET',
+        path=f'https://api.jwplayer.com/v2/sites/{site_id}/media/{media_id}/text_tracks/'
+    )
+    captions = []
+    for text_track in captions_response.json_body['text_tracks']:
+        captions.append(
+            {
+                'created': text_track['created'],
```
Comment (boyntoni) [inline examples/video_list_to_csv.py]: Would `delivery_url` be helpful?

File: examples/video_list_to_csv.py [hunk 3]
```diff
@@ -61,18 +61,41 @@ def make_csv(secret, site_id, path_to_csv=None, result_limit=1000, query_params=
...
     should_write_header = not os.path.isfile(path_to_csv)
     with open(path_to_csv, 'a+') as path_to_csv:
         # Only write columns to the csv which are specified above. Columns not specified are ignored.
         writer = csv.DictWriter(path_to_csv, fieldnames=desired_fields, extrasaction='ignore')
         if should_write_header:
             writer.writeheader()
         writer.writerows(videos)
+
+def get_captions(api_client, site_id, media_id):
+    captions_response = api_client.request(
```
Comment (boyntoni) [inline examples/video_list_to_csv.py]: In practice, likely not going to be much of an issue, but this request could fail due to rate limiting - or any other potential reason. Do we want to handle any of that here?

File: examples/video_list_to_csv.py [hunk 4]
```diff
@@ -61,18 +61,41 @@ def make_csv(secret, site_id, path_to_csv=None, result_limit=1000, query_params=
         for video in next_videos:
             csv_video = video["metadata"]
             csv_video["id"] = video["id"]
+            csv_video['duration'] = video['duration']
+            captions = get_captions(api_client=jwplatform_client, site_id=site_id, media_id=video["id"])
```
Comment (boyntoni) [inline examples/video_list_to_csv.py:63]: Looks like we might need to add `custom_params` here as well

File: examples/video_list_to_csv.py [hunk 5]
```diff
@@ -61,18 +61,41 @@ def make_csv(secret, site_id, path_to_csv=None, result_limit=1000, query_params=
...
+            csv_video['has_captions'] = True if len(captions) > 0 else False
```
Comment (boyntoni) [inline examples/video_list_to_csv.py]: It's a stylistic thing and I'm fine with this, but could be:

`bool(len(captions))`

File: examples/video_list_to_csv.py (hunk 2 above)
Comment (dannyjgibson) [reply examples/video_list_to_csv.py]: It could be added but I will leave that to a future request!

**Review by polishmatt — APPROVED** [review]

**Pull request merged**